java/vector/src/main/java/org/apache/arrow/vector/BitVector.java

```diff
@@ -126,7 +126,12 @@ public void setInitialCapacity(int valueCount) {
    */
   @Override
   public int getValueCapacity() {
-    return capAtMaxInt(validityBuffer.capacity() * 8);
+    return actualValueCapacity;
+  }
+
+  @Override
+  protected void refreshValueCapacity() {
+    actualValueCapacity = capAtMaxInt(validityBuffer.capacity() * 8);
```

Comment:
I am a little surprised here (I am aware that it was copied from the original implementation): the value capacity of a BitVector should also consider the capacity of the data buffer. So I think the original implementation has a bug here?

Reply:
Oh yes, that's true. They're two separate (bit)buffers. I'll fix the bug here.

Reply:
Hmm that means that this wasn't picked up on a unit test for some reason; handleSafe threads through the same implementation, looking at it.

Reply:
I guess you need to transfer the buffers or something with an intentionally broken source (if possible) to trigger this.